openquake/engine/celery_node_monitor.py

```diff
@@ -1,158 +0,0 @@
-# -*- coding: utf-8 -*-
-# vim: tabstop=4 shiftwidth=4 softtabstop=4
-#
-# Copyright (C) 2014-2016 GEM Foundation
-#
-# OpenQuake is free software: you can redistribute it and/or modify it
-# under the terms of the GNU Affero General Public License as published
-# by the Free Software Foundation, either version 3 of the License, or
-# (at your option) any later version.
-#
-# OpenQuake is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU Affero General Public License for more details.
-#
-# You should have received a copy of the GNU Affero General Public License
-# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
-
-import os
-import sys
-import time
-import signal
-import threading
-
-import celery.task.control
-
-from openquake.engine import logs
-from openquake.engine.utils import config
-from openquake.commonlib.valid import boolean
-from openquake.commonlib.oqvalidation import OqParam
-
-USE_CELERY = boolean(config.get('celery', 'use_celery'))
-
-
-class MasterKilled(KeyboardInterrupt):
-    """
-    Exception raised when a job is killed manually or aborted
-    by the `openquake.engine.engine.CeleryNodeMonitor`.
-    """
-    registered_handlers = False  # set when the signal handlers are registered
-
-    @classmethod
-    def handle_signal(cls, signum, _stack):
-        """
-        When a SIGTERM or a SIGABRT is received, raise the MasterKilled
-        exception with an appropriate error message.
-
-        :param int signum: the number of the received signal
-        :param _stack: the current frame object, ignored
-        """
-        if signum == signal.SIGTERM:
-            msg = 'The openquake master process was killed manually'
-        elif signum == signal.SIGABRT:
-            msg = ('The openquake master process was killed by the '
-                   'CeleryNodeMonitor because some node failed')
-        else:
-            msg = 'This should never happen'
-        raise cls(msg)
-
-    @classmethod
-    def register_handlers(cls):
-        """
-        Register the signal handlers for SIGTERM and SIGABRT
-        if they were not registered before.
-        """
-        if not cls.registered_handlers:  # called only once
-            signal.signal(signal.SIGTERM, cls.handle_signal)
-            signal.signal(signal.SIGABRT, cls.handle_signal)
-            cls.registered_handlers = True
-
-
-class CeleryNodeMonitor(object):
```

Comment:
An issue removing the celery monitor is that you don't get any feedback anymore if you start a calculation and celery is not started/not running. It seems that the calculation it's processing data in the pre execute phase, instead it's just waiting for a non existent celery consumer.